Rosella/src/main/java/me/hydos/rosella/device/VulkanQueue.java

```diff
@@ -0,0 +1,98 @@
+package me.hydos.rosella.device;
+
+import org.lwjgl.vulkan.*;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public final class VulkanQueue {
+
+    private final VkQueue queue;
+    private final int family;
+
+    private final Lock lock = new ReentrantLock();
+
+    public VulkanQueue(VkQueue queue, int family) {
+        this.queue = queue;
+        this.family = family;
+    }
+
+    public VkQueue getQueue() {
+        return this.queue;
+    }
+
+    public VkDevice getDevice() {
+        return this.queue.getDevice();
+    }
+
+    public int getQueueFamily() {
+        return this.family;
+    }
+
+    public int vkQueueSubmit(VkSubmitInfo submit, long fence) {
+        int result;
+        try {
+            this.lock.lock();
```

Comment:
should we make a function called lock() ? just a suggestion, nothing too important

Reply:
and maybe an unlock as well

Reply:
Done